Source/WebCore/loader/FrameLoader.h

```diff
@@ -352,6 +352,8 @@ class FrameLoader final : public CanMakeCheckedPtr<FrameLoader> {
     void loadItem(HistoryItem&, HistoryItem* fromItem, FrameLoadType, ShouldTreatAsContinuingLoad);
     HistoryItem* requestedHistoryItem() const { return m_requestedHistoryItem.get(); }
 
+    void updateURLAndHistory(const URL&, RefPtr<SerializedScriptValue>&& stateObject, NavigationHistoryBehavior historyHandling);
```

Comment:
I think we need a default for historyHandling:
"an optional [history handling behavior](https://html.spec.whatwg.org/multipage/browsing-the-web.html#history-handling-behavior) historyHandling (default "[replace](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigationhistorybehavior-replace)"),"

Reply:
Also could leave out the argument name since the type is unambiguous without a name.